src/validator/FetchEdgesValidator.cpp

```diff
@@ -284,5 +288,30 @@ std::string FetchEdgesValidator::buildRuntimeInput() {
     return inputVar_;
 }
 
+Expression* FetchEdgesValidator::emptyEdgeKeyFilter(const PlanNode *input) {
+    // _src != empty && _dst != empty && _rank != empty
+    DCHECK_GE(geColNames_.size(), 3);
+    auto *srcEmptyExpr = new RelationalExpression(
+        Expression::Kind::kRelNE,
+        new ConstantExpression(Value::kEmpty),
+        new VariablePropertyExpression(new std::string(input->outputVar()),
+                                       new std::string(geColNames_[0])));
+    auto *dstEmptyExpr = new RelationalExpression(
+        Expression::Kind::kRelNE,
+        new ConstantExpression(Value::kEmpty),
+        new VariablePropertyExpression(new std::string(input->outputVar()),
+                                       new std::string(geColNames_[1])));
+    auto *rankEmptyExpr = new RelationalExpression(
+        Expression::Kind::kRelNE,
+        new ConstantExpression(Value::kEmpty),
+        new VariablePropertyExpression(new std::string(input->outputVar()),
+                                       new std::string(geColNames_[2])));
```

Comment:
Why not provide a relation expression builder function to build these similar expressions?

Reply:
Had fixed.